cmake/modules/FindOrFetchSpdlog.cmake

```diff
@@ -10,7 +10,10 @@ if (NOT TARGET spdlog::spdlog)
         GIT_SHALLOW
         EXCLUDE_FROM_ALL
         SYSTEM
-        FIND_PACKAGE_ARGS NAMES spdlog
+		# The CMake option is not honored if spdlog is found via CMake config file
+		# However, our code doesn't compile when using libfmt as we're specializing templates for std::format
+		# see https://github.com/gabime/spdlog/issues/3467
+		#FIND_PACKAGE_ARGS NAMES spdlog
```

Comment:
@Krzmbrzl sounds like we need to do a 3-step dance:
- if TARGET spdlog::spdlog already available, still introspect it to detect if using std::format. If not, instist on FC.
- if TARGET spdlog::spdlog is not yet available, find_package(spdlog), introspect, if not using std::format insist on FC
- FC if needed.

Reply:
Introspecting seems somewhat fragile. I think I tend to stick with always using FetchContent for now (assuming that now top-level project has defined the target otherwise - in which case FC will cause a target conflict anyway). Going forward, I would switch to a different logging library instead.

Reply:
presumably can see if `SPDLOG_USE_STD_FORMAT` is among compile definitions of the spdlog target: https://github.com/gabime/spdlog/blob/v1.x/CMakeLists.txt#L287

not a big deal now but will probably need this to be adoptable by packagers